fiaas_deploy_daemon/deployer/kubernetes/deployment/deployer.py

```diff
@@ -82,7 +83,7 @@ def deploy(self, app_spec, selector, labels, besteffort_qos_is_required):
 
         automount_service_account_token = app_spec.admin_access
         init_containers = []
-        service_account_name = "default"
+        service_account_name = app_spec.name if self._enable_service_account_per_app is True else "default"
```

Comment:
```suggestion
        service_account_name = app_spec.name if self._enable_service_account_per_app else "default"
```
readability tip: Comparing booleans to values isn't necessary, you can just test them directly

Reply:
The tests fail if we don't have a value for that I think 🤔

Reply:
That was before.